source/MaterialXGenShader/Util.cpp

```diff
@@ -208,12 +246,11 @@ bool isTransparentSurface(ElementPtr element, const ShaderGenerator& shadergen)
                 const OutputPtr& output = outputs[0];
                 if (output->getType() == SURFACE_SHADER_TYPE_STRING)
                 {
-                    if (isTransparentShaderGraph(output, shadergen))
+                    OpaqueTestPairList opaqueInputList;
+                    if (isTransparentShaderGraph(output, target, node))
```

Comment:
At this point we know we are checking the functional graph for this node so only pass this node to check it's input interface names.